src/org/etools/j1939_84/model/SpnGroup.java

```diff
@@ -23,6 +24,15 @@ public boolean isSatisfied(Collection<Integer> supportedSpnValues) {
         return supportedSpnValues.stream().anyMatch(spns::contains);
     }
 
+    public boolean containsMultiple(Collection<Integer> supportedSpnValues) {
+        Collection<Integer> matchingSpns = new ArrayList<>();
+        if (spns.isEmpty()) {
```

Comment:
You don't need to even check for empty.  The logic below will return false if `spns` is empty.